cmd/site-replication.go

```diff
@@ -352,54 +392,78 @@ func (c *SiteReplicationSys) AddPeerClusters(ctx context.Context, sites []madmin
 
 	// Create a local service account.
 
-	// Generate a secret key for the service account.
+	// Generate a secret key for the service account if not created already.
 	var secretKey string
-	_, secretKey, err := auth.GenerateCredentials()
-	if err != nil {
-		return madmin.ReplicateAddStatus{}, SRError{
-			Cause: err,
-			Code:  ErrInternalError,
-		}
+	svcCred, _, err := globalIAMSys.getServiceAccount(ctx, siteReplicatorSvcAcc)
+	if err == nil {
+		secretKey = svcCred.SecretKey
 	}
-
-	svcCred, err := globalIAMSys.NewServiceAccount(ctx, sites[selfIdx].AccessKey, nil, newServiceAccountOpts{
-		accessKey: siteReplicatorSvcAcc,
-		secretKey: secretKey,
-	})
 	if err != nil {
```

Comment:
If the error is due to something other than the service account not existing, it could break existing SR by changing the password right?

Reply:
that's a good point - changed code to return an error if it is not due to non-existent service account